op-batcher/batcher/driver.go

```diff
@@ -542,92 +542,138 @@ func (l *BatchSubmitter) receiptsLoop(wg *sync.WaitGroup, receiptsCh chan txmgr.
 	l.Log.Info("receiptsLoop returning")
 }
 
-// throttlingLoop monitors the backlog in bytes we need to make available, and appropriately enables or disables
-// throttling of incoming data to prevent the backlog from growing too large. We ensure the engine currently in use
-// is always going to be reset to the proper throttling settings (even in the event of sequencer failover) by
-// binding to the ActiveSeqChanged channel as well as the pendingBytesUpdated channel.
-func (l *BatchSubmitter) throttlingLoop(wg *sync.WaitGroup, pendingBytesUpdated chan int64) {
+// singleEndpointThrottler handles throttling for a specific endpoint
+func (l *BatchSubmitter) singleEndpointThrottler(wg *sync.WaitGroup, throttleSignal chan struct{}, endpoint string) {
 	defer wg.Done()
-	l.Log.Info("Starting DA throttling loop")
+	l.Log.Info("Starting endpoint throttling loop", "endpoint", endpoint)
+
+	client, err := rpc.Dial(endpoint)
+	if err != nil {
+		// rpc.Dial returns an error if e.g. the URL is malformed
+		// If the server is unavailable, we will get an error when performing the first call
+		// and retries for that are handled below. Therefore  we don't need any retry logic here
+		l.Log.Error("Failed to Dial endpoint", "endpoint", endpoint, "err", err)
+		return
+	}
 
 	retryInterval := 10 * time.Second
 	retryTimer := time.NewTimer(retryInterval)
 	retryTimer.Stop()
 
-	updateParams := func(pendingBytes int64) {
+	updateParams := func() {
 		retryTimer.Stop()
+
 		ctx, cancel := context.WithTimeout(l.shutdownCtx, l.Config.NetworkTimeout)
 		defer cancel()
-		cl, err := l.EndpointProvider.EthClient(ctx)
-		if err != nil {
-			l.Log.Error("Can't reach sequencer execution RPC", "err", err)
-			return
-		}
+
+		l.mutex.Lock()
+		throttle := l.throttling
+		l.mutex.Unlock()
```

Comment:
I think this is a perfect use case for a simpler [`atomic.Bool`](https://pkg.go.dev/sync/atomic#Bool)

Reply:
[441042f](https://github.com/ethereum-optimism/optimism/pull/15603/commits/441042f6af32943df89598e2ce0812a2862484f5)